File: java/sample-springmvc-api/src/main/java/io/github/rajendrasatpute/samplespringmvcapi/client/OpenMeteoClient.java
package io.github.rajendrasatpute.samplespringmvcapi.client;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestClient;

@Component
public class OpenMeteoClient {

    @Autowired
    private RestClient openMeteoRestClient;

    public Object getWeather(String latitude, String longitude) {
        return openMeteoRestClient.get().uri(uriBuilder -> uriBuilder
                        .path("/forecast")
                        .queryParam("current", "temperature_2m,wind_speed_10m")
                        .queryParam("latitude", latitude)
                        .queryParam("longitude", longitude)
                        .queryParam("timezone", "IST")
                        .build())
                .retrieve()
                .body(Object.class);
    }

}

File: python/sample-fastapi-api/app/clients/openmeteo_client.py
import httpx

URL = 'https://api.open-meteo.com/v1/forecast?current=temperature_2m,wind_speed_10m&timezone=IST'


async def get_weather_async(latitude: str, longitude: str):
    params = {'latitude': latitude, 'longitude': longitude}
    async with httpx.AsyncClient() as client:
        response = await client.get(URL, params=params)
        return response.json() if response.status_code == 200 else {}
